integration/spark2/src/main/scala/org/apache/carbondata/spark/rdd/AggregateDataMapCompactor.scala

```diff
@@ -47,7 +47,15 @@ class AggregateDataMapCompactor(carbonLoadModel: CarbonLoadModel,
   override def executeCompaction(): Unit = {
     val carbonTable = carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable
     val loadMetaDataDetails = identifySegmentsToBeMerged()
-    val segments = loadMetaDataDetails.asScala.map(_.getLoadName)
+    // If segmentFile name is specified in load details then segment is for partition table
+    // therefore the segment file name should be loadName#segmentFileName.segment
+    val segments = loadMetaDataDetails.asScala.map {
+      loadDetail => if (loadDetail.getSegmentFile != null) {
+        loadDetail.getLoadName + "#" + loadDetail.getSegmentFile
+      } else {
+        loadDetail.getLoadName
+      }
```

Comment:
Just use new Segment(loadDetail.getLoadName, loadDetail.getSegmentFile).toString instead of doing this

Reply:
ok